### ISSUE TITLE
referenced types actually exist

### ISSUE THREAD

**Issue by anz-gordonj7** [anz-bank/sysl#466]
```
!type Foo:
    value <: X.Y
```

X.Y needs to exist

**Comment by anzdaddy** [comment]
`value <: X.Y` normally indicates that field `value` has a foreign key relationship to field `Y` in table `X`. It should also be possible for `Y` to be a type and `X` to be an app or type containing `Y`.

**Comment by anz-gordonj7** [comment]
> `value <: X.Y` normally indicates that field `value` has a foreign key relationship to field `Y` in table `X`. It should also be possible for `Y` to be a type and `X` to be an app or type containing `Y`.

Yes, the validator needs to determine what X.Y means, a !type should never have the behaviour you're talking about but a !table could. Could we maybe move that foreign key to an attribute?
